core/src/main/java/org/infinispan/commands/irac/IracRequestStateCommand.java

```diff
@@ -1,41 +1,45 @@
 package org.infinispan.commands.irac;
 
-import java.io.IOException;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
 import java.util.concurrent.CompletionStage;
 
 import org.infinispan.commands.remote.BaseRpcCommand;
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
 import org.infinispan.commons.util.IntSet;
-import org.infinispan.commons.util.IntSetsExternalization;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
 import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.marshall.protostream.impl.WrappedMessages;
+import org.infinispan.protostream.WrappedMessage;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
 import org.infinispan.util.ByteString;
-import org.infinispan.commons.util.concurrent.CompletableFutures;
 
 /**
  * Requests the state for a given segments.
  *
  * @author Pedro Ruivo
  * @since 11.0
  */
+@ProtoTypeId(ProtoStreamTypeIds.IRAC_REQUEST_STATE_COMMAND)
 public class IracRequestStateCommand extends BaseRpcCommand {
 
    public static final byte COMMAND_ID = 121;
 
    private IntSet segments;
```

Comment:
maybe worth to change this to `SmallIntSet` to avoid wrapping?

Reply:
Can we guarantee that a `SmallIntSet` is always applicable? The `CommandsFactory` method signature is `IracRequestStateCommand buildIracRequestStateCommand(IntSet segments) `

Reply:
yes, I tracked it down here https://github.com/infinispan/infinispan/blob/e9d227d6df1e9a1a5034737dbab2c37b2c82a7d1/core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java#L216

This is only for state transfer, so probably not worth changing the entire methods chain.